flink-core/src/test/java/org/apache/flink/core/fs/SafetyNetCloseableRegistryTest.java

```diff
@@ -198,4 +198,29 @@ public void testReaperThreadSpawnAndStop() throws Exception {
 		}
 		Assert.assertFalse(SafetyNetCloseableRegistry.isReaperThreadRunning());
 	}
+
+	/**
+	 * Test whether failure to start thread in {@link SafetyNetCloseableRegistry}
+	 * constructor can lead to failure of subsequent state check.
+	 */
+	@Test
+	public void testReaperThreadStartFailed() throws Exception {
+
+		try {
+			new SafetyNetCloseableRegistry(() -> new OutOfMemoryReaperThread());
+		} catch (java.lang.OutOfMemoryError error) {
+		}
+
```

Comment:
We can add `isReaperThreadRunning` checks after the failed creation and the succeeded creation. 
This helps to verify that a reaper thread is really created and running.

Reply:
OK, I will add the check.